be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -71,35 +71,33 @@ Status LakePrimaryIndex::_do_lake_load(Tablet* tablet, const TabletMetadata& met
     if (tablet->get_enable_persistent_index(base_version) && (fix_size <= 128)) {
         DCHECK(_persistent_index == nullptr);
 
-        // Even if `enable_persistent_index` is enabled,
-        // it may not take effect because `storage_root_path` is not set
-        if (StorageEngine::instance()->is_lake_persistent_index_dir_inited()) {
-            auto persistent_index_type = tablet->get_persistent_index_type(base_version);
-            if (persistent_index_type.ok()) {
-                switch (persistent_index_type.value()) {
-                case PersistentIndexTypePB::LOCAL: {
-                    std::string path = strings::Substitute(
-                            "$0/$1/",
-                            StorageEngine::instance()->get_persistent_index_store()->get_persistent_index_path(),
-                            tablet->id());
-
-                    RETURN_IF_ERROR(
-                            StorageEngine::instance()->get_persistent_index_store()->create_dir_if_path_not_exists(
-                                    path));
-                    _persistent_index = std::make_unique<LakeLocalPersistentIndex>(path);
-                    return ((LakeLocalPersistentIndex*)_persistent_index.get())
-                            ->load_from_lake_tablet(tablet, metadata, base_version, builder);
-                }
-                default:
-                    LOG(WARNING) << "only support LOCAL lake_persistend_index_type for now";
-                    return Status::InternalError("only support LOCAL lake_persistend_index_type for now");
+        auto persistent_index_type = tablet->get_persistent_index_type(base_version);
+        if (persistent_index_type.ok()) {
+            switch (persistent_index_type.value()) {
+            case PersistentIndexTypePB::LOCAL: {
+                // Even if `enable_persistent_index` is enabled,
+                // it may not take effect if is as compute node.
+                if (StorageEngine::instance()->is_as_cn()) {
+                    LOG(WARNING) << "lake_persistent_index_type of LOCAL will not take effect when as cn";
+                    return Status::InternalError("lake_persistent_index_type of LOCAL will not take effect when as cn");
```

Comment:
This error message is not user friendly enough, and if it does happen, what does the user have to do to fix the error?

Reply:
Currently, local persistent index does not allow any cn nodes without storage_path, and an error will be throwed when creating table if there are cn nodes without storage_path and enable `persistent_index`. This situation may happen when adding node,  Dropping the cn node can fix。But persistent index only need valid storage path，i will change the condititon to judge whether there are storage paths， so user can fix the error by config `storage_path`